### PR TITLE
cleanup(otel): neater span matchers

### PR DESCRIPTION
**Pull request by dbolduc**
In response to @devbww's comments: https://github.com/googleapis/google-cloud-cpp/pull/10501#pullrequestreview-1244622069

Fix up the error output for span matchers. For example, this test:

```
TEST(OpenTelemetry, Fail) {
  auto span_catcher = InstallSpanCatcher();
  auto provider = opentelemetry::trace::Provider::GetTracerProvider();
  auto tracer = provider->GetTracer("bad-scope-name", "bad-scope-version");

  auto s = tracer->StartSpan("bad-span-name");
  s->End();

  auto spans = span_catcher->GetSpans();
  auto& span = spans[0];
  EXPECT_THAT(span, SpanHasInstrumentationScope());
  EXPECT_THAT(span, SpanKindIsClient());
  EXPECT_THAT(span, SpanNamed("expected-span-name"));
}
```

... produces the following error output:

```
[ RUN      ] OpenTelemetry.Fail
/workspace/google/cloud/internal/opentelemetry_test.cc:98: Failure
Value of: span
Expected: has instrumentation scope (name: gcloud-cpp | version: v2.7.0-rc)
  Actual: (ptr = 0x1964b40, value = 304-byte object <18-A3 45-00 00-00 00-00 C4-2D 1B-29 E4-D8 DC-9B 10-35 DF-F3 E9-EA B8-80 00-6D CB-E3 30-C5 71-32 01-00 00-00 00-00 00-00 C0-79 45-00 00-00 00-00 40-3D 96-01 00-00 00-00 30-38 96-01 00-00 00-00 ... 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 70-51 96-01 00-00 00-00 F0-40 96-01 00-00 00-00>), has instrumentation scope (name: bad-scope-name | version: bad-scope-version)
/workspace/google/cloud/internal/opentelemetry_test.cc:99: Failure
Value of: span
Expected: has span kind: CLIENT
  Actual: (ptr = 0x1964b40, value = 304-byte object <18-A3 45-00 00-00 00-00 C4-2D 1B-29 E4-D8 DC-9B 10-35 DF-F3 E9-EA B8-80 00-6D CB-E3 30-C5 71-32 01-00 00-00 00-00 00-00 C0-79 45-00 00-00 00-00 40-3D 96-01 00-00 00-00 30-38 96-01 00-00 00-00 ... 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 70-51 96-01 00-00 00-00 F0-40 96-01 00-00 00-00>), has span kind: INTERNAL
/workspace/google/cloud/internal/opentelemetry_test.cc:100: Failure
Value of: span
Expected: has name: expected-span-name
  Actual: (ptr = 0x1964b40, value = 304-byte object <18-A3 45-00 00-00 00-00 C4-2D 1B-29 E4-D8 DC-9B 10-35 DF-F3 E9-EA B8-80 00-6D CB-E3 30-C5 71-32 01-00 00-00 00-00 00-00 C0-79 45-00 00-00 00-00 40-3D 96-01 00-00 00-00 30-38 96-01 00-00 00-00 ... 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 70-51 96-01 00-00 00-00 F0-40 96-01 00-00 00-00>), has name: bad-span-name
[  FAILED  ] OpenTelemetry.Fail (0 ms)
```

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10554)

### CODE REVIEW

File: google/cloud/testing_util/opentelemetry_matchers.h [hunk 1]
```diff
@@ -32,55 +32,30 @@ namespace testing_util {
...
-      case opentelemetry::trace::SpanKind::kProducer:
-        return "PRODUCER";
-      case opentelemetry::trace::SpanKind::kConsumer:
-      default:
-        return "UNKNOWN";
-    }
-  };
-
+MATCHER(SpanKindIsClient,
+        "has span kind: " + ToString(opentelemetry::trace::SpanKind::kClient)) {
```
Comment (devbww) [inline google/cloud/testing_util/opentelemetry_matchers.h:49]: I might `s/has span kind/had kind/` here and below, to match the other matchers.

**Review by devbww — APPROVED** [review]
Nice.

File: google/cloud/testing_util/opentelemetry_matchers.h (hunk 1 above)
Comment (dbolduc) [reply google/cloud/testing_util/opentelemetry_matchers.h]: Updated, but I kept the present tense

Comment (devbww) [reply google/cloud/testing_util/opentelemetry_matchers.h]: Sorry ... "had" was a slip of the finger.

**Pull request merged**